lxd/certificates.go

```diff
@@ -354,6 +354,50 @@ func clusterMemberJoinTokenValid(d *Daemon, r *http.Request, projectName string,
 	return nil, nil
 }
 
+// certificateTokenValid searches for certificate token that matches the add token provided.
+// Returns matching operation if found and cancels the operation, otherwise returns nil.
+func certificateTokenValid(d *Daemon, r *http.Request, projectName string, addToken *api.CertificateAddToken) (*api.Operation, error) {
+	ops, err := operationsGetByType(d, r, projectName, db.OperationCertificateAddToken)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed getting certificate token operations")
+	}
+
+	var foundOp *api.Operation
+	for _, op := range ops {
+		if op.StatusCode != api.Running {
+			continue // Tokens are single use, so if cancelled but not deleted yet its not available.
+		}
+
+		opSecret, ok := op.Metadata["secret"]
+		if !ok {
+			continue
+		}
+
+		opFingerprint, ok := op.Metadata["fingerprint"]
+		if !ok {
+			continue
+		}
+
```

Comment:
No need to validate the fingerprint, it's a field that's useful to the client to connect but if the server gets its certificate replaced it shouldn't make all tokens invalid.